pandas/core/indexes/datetimelike.py

```diff
@@ -719,33 +719,29 @@ def intersection(self, other, sort=False):
         """
         self._validate_sort_keyword(sort)
         self._assert_can_do_setop(other)
-        res_name = get_op_result_name(self, other)
 
         if self.equals(other):
             return self._get_reconciled_name_object(other)
 
         if len(self) == 0:
-            return self.copy()
+            return self.copy()._get_reconciled_name_object(other)
```

Comment:
in the base class you got rid of the `_get_reconciled_name_object` usage.  why going the other direction here?

Reply:
Still keeping this around for the rare case where you want to return the same values but with a possible name change. In base.py it only shows up in the one `intersection` return clause (and not in `union` because of the `union`/`_union` split). It is interesting though that the version of `intersection` in datetimelike.py has two more quick return clauses than the base class.